assets/js/payment-method-extensions/payment-methods/stripe/payment-request/index.js

```diff
@@ -0,0 +1,41 @@
+/**
+ * Internal dependencies
+ */
+import { PAYMENT_METHOD_NAME } from './constants';
+import { PaymentRequestExpress } from './payment-request-express';
+import { applePayImage } from './apple-pay-preview';
+import { loadStripe } from '../stripe-utils';
+
+const ApplePayPreview = () => <img src={ applePayImage } alt="" />;
+
+const stripePromise = loadStripe();
+
+const PaymentRequestPaymentMethod = {
+	id: PAYMENT_METHOD_NAME,
+	content: <PaymentRequestExpress stripe={ stripePromise } />,
+	edit: <ApplePayPreview />,
+	canMakePayment: stripePromise.then( ( stripe ) => {
+		if ( stripe === null ) {
+			return false;
+		}
+		// do a test payment request to check if payment request payment can be
+		// done.
+		// @todo, initial country and currency needs to server.
+		const paymentRequest = stripe.paymentRequest( {
+			total: {
+				label: 'Test total',
+				amount: 1000,
+			},
+			country: 'CA',
+			currency: 'cad',
+		} );
```

Comment:
This needs to be handled in a follow-up so that `country` and `currency` are provided by the server.

Reply:
This can either use store settings (base country, currency) or we can add some extra data to checkout endpoint (billing country, with base country fallback, and order currency)